### ISSUE TITLE
Panic.log is read only after rebooting the VM (Windows Server 2022)

### ISSUE THREAD

**Issue by Sven-Niehus** [docker/for-win#12358]
- [X] I have tried with the latest version of Docker Desktop
  - [X] I have tried disabling enabled experimental features
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID:

### Actual behavior
Docker fails to start after a reboot of the machine docker is running on, because the file "panic.log" in the docker data root is marked as read-only.

### Expected behavior
Docker should continue to work after rebooting the machine.

### Information

  - Windows Version: Windows Server 2022 Standard (21H2 OS Build 20348.288)
  - Docker Desktop Version: 
  ```
  Client:
   Version:           20.10.10
   API version:       1.41
   Go version:        go1.16.9
   Git commit:        b485636
   Built:             Mon Oct 25 07:47:53 2021
   OS/Arch:           windows/amd64
   Context:           default
   Experimental:      true
  
  Server: Docker Engine - Community
   Engine:
    Version:          20.10.10
    API version:      1.41 (minimum version 1.24)
    Go version:       go1.16.9
    Git commit:       e2f740d
    Built:            Mon Oct 25 07:43:13 2021
    OS/Arch:          windows/amd64
    Experimental:     false
  ```
  - WSL2 or Hyper-V backend? Hyper-V
  - Are you running inside a virtualized Windows e.g. on a cloud server or a VM: Yes, the machine I'm running docker on is virtualized with Hyper-V
  - docker info:
  ```
  Client:
   Context:    default
   Debug Mode: false
   Plugins:
    app: Docker App (Docker Inc., v0.9.1-beta3)
    buildx: Build with BuildKit (Docker Inc., v0.6.3-docker)
  
  Server:
   Containers: 0
    Running: 0
    Paused: 0
    Stopped: 0
   Images: 1
   Server Version: 20.10.10
   Storage Driver: windowsfilter
    Windows:
   Logging Driver: json-file
   Plugins:
    Volume: local
    Network: ics internal l2bridge l2tunnel nat null overlay private transparent
    Log: awslogs etwlogs fluentd gcplogs gelf json-file local logentries splunk syslog
   Swarm: inactive
   Default Isolation: process
   Kernel Version: 10.0 20348 (20348.1.amd64fre.fe_release.210507-1500)
   Operating System: Windows Server 2022 Standard Evaluation Version 2009 (OS Build 20348.288)
   OSType: windows
   Architecture: x86_64
   CPUs: 4
   Total Memory: 4.999GiB
   Name: __
   ID: OPJI:6U7B:V44U:CDEY:UMO7:HV6S:6RTB:ILZK:PCGC:EZZC:2L3P:4KHT
   Docker Root Dir: C:\tmp\docker
   Debug Mode: false
   Registry: https://index.docker.io/v1/
   Labels:
   Experimental: false
   Insecure Registries:
    127.0.0.0/8
   Live Restore Enabled: false
   Product License: Community Engine
  ```
 - systeminfo:
 ```
  Host Name:                 __
  OS Name:                   Microsoft Windows Server 2022 Standard Evaluation
  OS Version:                10.0.20348 N/A Build 20348
  OS Manufacturer:           Microsoft Corporation
  OS Configuration:          Member Server
  OS Build Type:             Multiprocessor Free
  Registered Owner:
  Registered Organization:
  Product ID:                00454-40000-00001-AA405
  Original Install Date:     11/4/2021, 1:43:40 PM
  System Boot Time:          11/4/2021, 2:40:12 PM
  System Manufacturer:       Microsoft Corporation
  System Model:              Virtual Machine
  System Type:               x64-based PC
  Processor(s):              1 Processor(s) Installed.
                             [01]: Intel64 Family 6 Model 85 Stepping 4 GenuineIntel ~2095 Mhz
  BIOS Version:              Microsoft Corporation Hyper-V UEFI Release v4.0, 12/17/2019
  Windows Directory:         C:\Windows
  System Directory:          C:\Windows\system32
  Boot Device:               \Device\HarddiskVolume2
  System Locale:             en-us;English (United States)
  Input Locale:              de;German (Germany)
  Time Zone:                 (UTC+01:00) Amsterdam, Berlin, Bern, Rome, Stockholm, Vienna
  Total Physical Memory:     5,119 MB
  Available Physical Memory: 2,671 MB
  Virtual Memory: Max Size:  6,655 MB
  Virtual Memory: Available: 4,410 MB
  Virtual Memory: In Use:    2,245 MB
  Page File Location(s):     C:\pagefile.sys
  Domain:                    __
  Logon Server:              __
  Hotfix(s):                 3 Hotfix(s) Installed.
                             [01]: KB5006364
                             [02]: KB5007205
                             [03]: KB5006752
  Network Card(s):           1 NIC(s) Installed.
                             [01]: Microsoft Hyper-V Network Adapter
                                   Connection Name: Ethernet
                                   DHCP Enabled:    Yes
                                   DHCP Server:     __
                                   IP address(es)
                                   [01]: __
                                   [02]: __
 ```

### Steps to reproduce the behavior

  1. On a fresh install of Windows Server 2022 install all available updates
  2. Install docker via latest zip file from https://download.docker.com/win/static/stable/x86_64/
  3. Register docker as a system service with ".\dockerd.exe --register-service"
  4. Restart the machine
  5. Now the panic.log file is read-only 
![image](https://user-images.githubusercontent.com/9822857/141075682-0e03cfc1-b5a2-4b09-907a-52c9b189d16e.png)
 and docker fails to start with the following error in the event log
![image](https://user-images.githubusercontent.com/9822857/141075929-8c69e2bb-b412-40be-8444-788cbce4fa09.png)

**Comment by thaJeztah** [comment]
Thanks for reporting; this looks like a duplicate of https://github.com/docker/for-win/issues/11899 (which mentions some workarounds), but note that this issue tracker is for Docker Desktop, which is not supported on Windows Server, not for standalone installs of the docker engine on windows.

Docker Desktop 4.1.0 and up already contains a for for this issue. For standalone installs of the docker engine on Windows, a fix was merged in the upstream engine code; https://github.com/moby/moby/pull/42984, which has been back ported to the 20.10 release branch (https://github.com/moby/moby/pull/42987), and will be included in an upcoming 20.10.11 release of the Docker Engine.

Let me close this because of the above, but feel free to continue the conversation
